### PR TITLE
Promo card CTA: remove invalid prop on DOM element

### PR DESCRIPTION
**Pull request by monsieur-z**
#### Changes proposed in this Pull Request

In `PromoCardCta`, a lingering `selfTarget` prop passed as an attribute to an HTML element triggered a React warning in the console (`selfTarget` not being a valid attribute).

image

This commit makes sure it's not passed anymore.

#### Testing instructions

1. Make sure you've got a WP site with a Free plan
2. Download the PR and boot Calypso
3. Make sure to enable Jetpack sections with the query param `?flags=jetpack/features-section`
4. Select the free plan site
5. Go to the _Jetpack > Backup_ section
6. Check there's no React error in the console
7. Click on the _Upgrade to business plan_ button and check that you're still in the same tab
8. Repeat the operation with the _Jetpack > Scan_ section

### CODE REVIEW

File: client/components/promo-section/promo-card/cta.tsx [hunk 1]
```diff
@@ -50,25 +50,28 @@ export interface Props {
 	isPrimary?: boolean;
 }
 
-function isCtaButton( cta: Cta ): cta is CtaButton {
+function isCtaButton( cta: Cta ): boolean {
 	return undefined !== ( cta as CtaButton ).text;
 }
 
-function isCtaAction( action: any ): action is CtaAction {
+function isCtaAction( action: any ): boolean {
```
Comment (ChaosExAnima) [inline client/components/promo-section/promo-card/cta.tsx:57]: This argument isn't any, it is a union type of `URL | ClickCallback | CtaAction` going by the code above.

Additionally, `action is CtaAction` is a very specific return type that indicates to TS that a given action is a CtaAction. With this code, I see many of the following or similar:
```
Property 'selfTarget' does not exist on type 'string | CtaAction | ClickCallback'.
```

**Review by ChaosExAnima — CHANGES_REQUESTED** [review]
This code has multiple side effects and is causing a lot of errors when I review it with TS warnings in my environment.

I think this PR should be simplified to target just the removal of `selfTarget` instead of declaring more constants and changing the inner workings. This component could do with a refactor in general honestly, but that should be a different PR imo.

**Comment by monsieur-z** [comment]
Some Git mishap here. Let me open a new PR instead. I'll link this one.

File: apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/components/nav-sidebar/index.tsx [hunk 2]
```diff
@@ -130,23 +137,81 @@ export default function WpcomBlockEditorNavSidebar() {
...
+	} );
+
 	return (
 		<li>
-			<div>{ item.title.rendered }</div>
-			<pre>{ `/${ item.slug }/` }</pre>
+			<Button className={ buttonClasses }>
+				<div className="wpcom-block-editor-nav-sidebar__title-container">
+					<div className={ titleClasses }>
+						{ item.title?.raw || __( 'Untitled', 'full-site-editing' ) }
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/components/nav-sidebar/index.tsx:158]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **40** times:
`translate( 'Untitled', { context: 'post title for posts with no title'} )` *ES Score: 16*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/jetpack-self-target-warning)

**Comment by monsieur-z** [comment]
Follow up in #43483 43483

**Pull request closed without merging**